core/src/cgcloud/core/box.py

```diff
@@ -406,8 +552,47 @@ def create( self, wait_ready=True, cluster_ordinal=0 ):
                 result.append( box )
         except StopIteration:
             pass
+
         return result
 
+    def _spot_create( self ):
+        """
+        A request for multiple instances actually creates many requests for 1 instance each,
+        which then become reservations with 1 instance each.
+
+        :rtype: list[boto.ec2.instance.Reservation]
+        """
+        requests = self.ctx.ec2.request_spot_instances( image_id=self.image_id,
+                                                        **self.instance_creation_args )
+        instances = [ ]
+
+        instance_ids = set()
+        try:
+            while True:
+                updatedRequests = self.ctx.ec2.get_all_spot_instance_requests(
+                    request_ids=[ request.id for request in requests] )
+                for request in updatedRequests:
+                    if request.status.code == 'fulfilled':
+                        # The get_all_instances always() method returns a list, though we know
+                        # there is only 1 instance.
+                        instance_ids.add(request.instance_id )
+                if len(instance_ids) < len(requests):
```

Comment:
For future reference, starting with a full set and removing elements is more elegant than starting with an empty set and adding elements.